pom.xml

```diff
@@ -247,8 +278,8 @@
           <artifactId>maven-compiler-plugin</artifactId>
           <version>3.11.0</version>
           <configuration>
-            <source>1.8</source>
-            <target>1.8</target>
+            <source>${compileSource}</source>
+            <target>${compileSource}</target>
```

Comment:
Oh, this is incorrect, we still need to generate the java 8 bytecode. For JDK version higher than 1.8, we could use '--release 8' to force it genereate the java 8 byte code and also use the symbols from JDK8.

The only difference is hbase-unsafe, because it depends on jdk.unsupported, which is not supported at compile time in JDK17, so we need to use JDK8 to compile it, and set source and target here.